pkg/controllers/gameservers/util.go

```diff
@@ -198,7 +196,7 @@ func buildPod(gs *carrierv1alpha1.GameServer) (*corev1.Pod, error) {
 	}
 
 	podObjectMeta(gs, pod)
-	if isHostPortNetwork(&gs.Spec) || len(findPorts(gs)) > 0 {
+	if len(findPorts(gs)) > 0 {
```

Comment:
will add a type for hostNetwork type

Reply:
For non-hostPort container, we can also set `Pod.Spec....ContainerPort`